packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx

```diff
@@ -738,14 +743,18 @@ describe('<DataGrid /> - Rows', () => {
           '.MuiDataGrid-virtualScrollerContent',
         )!;
         await act(() => Promise.resolve());
-        clock.runToLast();
+        clock.runToLast(); // Run timers in the mocked ResizeObserver
+        clock.runToLast(); // Run the debounce timeout that triggers the rerender
         expect(virtualScrollerContent).toHaveInlineStyle({
           width: 'auto',
           height: `${Math.floor(expectedHeight)}px`,
         });
       });
 
-      it('should position correctly the render zone when the 2nd page has less rows than the 1st page', async () => {
+      it('should position correctly the render zone when the 2nd page has less rows than the 1st page', async function test() {
+        if (/edg/i.test(window.navigator.userAgent)) {
+          this.skip(); // FIXME: We need a waitFor that works with fake clock
```

Comment:
@m4theushw 
This test also fails in Chrome and Firefox sometimes:

https://app.circleci.com/pipelines/github/mui/mui-x/21944/workflows/5b674c92-8ba9-427e-a902-d1a516513cd6/jobs/126695
```
Firefox 78.0 (Windows 10) <DataGrid /> - Rows prop: getRowHeight dynamic row height should position correctly the render zone when the 2nd page has less rows than the 1st page FAILED
	expected inline style of <div …(2)>…(1)</div> did not match
	Expected:
	{
	  "transform": "translate3d(0px, 0px, 0px)"
	}
	Actual:
	{
	  "transform": "translate3d(0px, 988px, 0px)"
	}
```

https://app.circleci.com/pipelines/github/mui/mui-x/21943/workflows/4ec09158-5c2c-4fb8-b994-52816ab953e1/jobs/126682
```
Chrome 87.0.4280.66 (Mac OS 10.15.6) <DataGrid /> - Rows prop: getRowHeight dynamic row height should position correctly the render zone when the 2nd page has less rows than the 1st page FAILED
	AssertionError: expected inline style of <div …(2)></div> did not match
	Expected:
	{
	  "transform": "translate3d(0px, 0px, 0px)"
	}
	Actual:
	{
	  "transform": "translate3d(0px, 1040px, 0px)"
	}
```

I can reproduce it locally using `Chrome` instead of `ChromeHeadless`.
Test fails **only** when browser window is in background.
```diff
diff --git a/test/karma.conf.js b/test/karma.conf.js
index 2ea932985..f40b51137 100644
--- a/test/karma.conf.js
+++ b/test/karma.conf.js
@@ -44,7 +44,7 @@ const MAX_CIRCLE_CI_CONCURRENCY = 83;
 module.exports = function setKarmaConfig(config) {
   const baseConfig = {
     basePath: '../',
-    browsers: ['chromeHeadless'],
+    browsers: ['Chrome'],
     browserDisconnectTimeout: 3 * 60 * 1000, // default 2000
     browserDisconnectTolerance: 1, // default 0
     browserNoActivityTimeout: 6 * 60 * 1000, // default 10000
```

Any ideas?

Reply:
I don't have any clue of why they fail. When I did this PR, I only had problems with Edge so ignoring this browser in this specific test solved the problem. https://app.circleci.com/pipelines/github/mui/mui-x/21728/workflows/12ccc593-a493-414b-9484-79cb96b8a991/jobs/125448 is from [a7eef9e](https://github.com/mui/mui-x/pull/5707/commits/a7eef9eb4006af4748481350556443f6fbc96b28)

I think we'll need to skip all tests related to dynamic row height since they are too flacky.